sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -215,7 +253,13 @@ private[sql] class RocksDBStateStoreProvider
       (keySchema.length > numColsPrefixKey), "The number of columns in the key must be " +
       "greater than the number of columns for prefix key!")
 
-    this.encoder = RocksDBStateEncoder.getEncoder(keySchema, valueSchema, numColsPrefixKey)
+    if (useMultipleValuesPerKey) {
```

Comment:
After my refactoring change - we can get rid of this check I guess

Reply:
Yes, we have the option to get rid of this check post the key/value encoder refactoring.

We can keep it if we want to support multiple values for a single key in default column family. It would make the supported operations consistent b/w default and other column families. But it can be added later, I am not strongly opinionated on this atm.